### ISSUE TITLE
[Bug]: The variable 'step_rew_funcs' is not defined in BaseReward class.

### ISSUE THREAD

**Issue by han-x** [OpenRL-Lab/openrl#30]
### 🐛 Bug

The variable 'step_rew_funcs' is not defined in BaseReward class. In addition, the 'step' function of the RewardWrapper class should also be modified accordingly.
'
        if extra_data:
            extra_data.update({"actions": action})
            extra_data.update({"obs": obs})
            extra_data.update({"rewards": rewards})
            extra_data.update({"dones": dones})
            extra_data.update({"infos": infos})
            rewards, new_infos = self.reward_class.step_reward(data=extra_data)

            num_envs = len(infos)
            for i in range(num_envs):
                infos[i].update(new_infos[i])
'

### To Reproduce

```python
from openrl import ...

```

### Relevant log output / Error message

```shell
Traceback (most recent call last):
  File "/data/workspace/openrl/examples/cartpole/train_ppo.py", line 39, in <module>
    agent = train()
  File "/data/workspace/openrl/examples/cartpole/train_ppo.py", line 17, in train
    agent.train(total_time_steps=20000)
  File "/data/workspace/openrl/openrl/runners/common/ppo_agent.py", line 134, in train
    driver.run()
  File "/data/workspace/openrl/openrl/drivers/onpolicy_driver.py", line 227, in run
    self._inner_loop()
  File "/data/workspace/openrl/openrl/drivers/onpolicy_driver.py", line 112, in _inner_loop
    rollout_infos = self.actor_rollout()
  File "/data/workspace/openrl/openrl/drivers/onpolicy_driver.py", line 189, in actor_rollout
    obs, rewards, dones, infos = self.envs.step(actions, extra_data)
  File "/data/workspace/openrl/openrl/envs/vec_env/wrappers/vec_monitor_wrapper.py", line 37, in step
    returns = self.env.step(action, extra_data)
  File "/data/workspace/openrl/openrl/envs/vec_env/wrappers/reward_wrapper.py", line 50, in step
    infos[i].update(new_infos[i])
IndexError: list index out of range
```

### Checklist

- [X] I have checked that there is no similar [issues](https://github.com/OpenRL-Lab/openrl/issues) in the repo
- [X] I have read the [documentation](https://openrl-docs.readthedocs.io/)
- [X] I have provided a minimal working example to reproduce the bug
- [X] I have version numbers, operating system and environment, where applicable

**Comment by han-x** [comment]
Maybe you guys should run all the examples before releasing it.

**Comment by huangshiyu13** [comment]
bug fixed in #29
